sdk/search/Azure.Search.Documents/src/Generated/Models/Vector.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System.Collections.Generic;
+using Azure.Core;
+
+namespace Azure.Search.Documents.Models
+{
+    /// <summary> The query parameters for vector and hybrid search queries. </summary>
+    public partial class Vector
+    {
+        /// <summary> Initializes a new instance of Vector. </summary>
+        public Vector()
+        {
+            Value = new ChangeTrackingList<float>();
+        }
+
+        /// <summary> Initializes a new instance of Vector. </summary>
+        /// <param name="value"> The vector representation of a search query. </param>
+        /// <param name="k"> Number of nearest neighbors to return as top hits. </param>
+        /// <param name="fields"> Vector Fields of type Collection(Edm.Single) to be included in the vector searched. </param>
+        internal Vector(IList<float> value, int? k, string fields)
+        {
+            Value = value;
+            K = k;
+            Fields = fields;
+        }
+
+        /// <summary> The vector representation of a search query. </summary>
+        public IList<float> Value { get; }
```

Comment:
@pallavit, @tg-msft - Can we add following C# code gen customization?
* Update the type of Vector.Value(embedding) from `IList<float>` to `IReadOnlyList<float>` to match the OpenAI embedding [type](https://github.com/Azure/azure-sdk-for-net/blob/97277e64fd4a4abfdb8fbe56f6c0e40f1eef7dbd/sdk/openai/Azure.AI.OpenAI/api/Azure.AI.OpenAI.netstandard2.0.cs#L110), so that customers won't have to call unnecessarily expensive [ToList()](https://github.com/ShivangiReja/DemoVectorSearch/blob/main/DemoVectorSearch/Program.cs#L62) operation.
* Add setter for `Value`, so customers won't have to use the [foreach loop](https://github.com/ShivangiReja/DemoVectorSearch/blob/main/DemoVectorSearch/Program.cs#LL104C43-L104C43) to add the vector values.

Reply:
Settable collections are against the .NET guidelines. They can use the collection initializer pattern:

```c#
Vector v = new()
{
  Value =
  {
    0.1,
    0.9,
  },
};
```

Is this in a response-only model? If so, it should be an `IReadOnlyList<float>` already, or is that not in the newer DPG code, @m-nash? Or will marking the property read-only in TypeSpec/swagger help here if it is read-only?

Reply:
This is a little different than a typical .NET collection.  They're immutable and we should change the type to `IReadOnlyList<float>` and add a setter.